src/main.rs

```diff
@@ -139,3 +143,93 @@ async fn accept_form(
     // Show index page
     show_index(Some(&msg))
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    use axum::body::Body;
+    use axum::http::header::CONTENT_TYPE;
+    use axum::http::{Request, StatusCode};
+    use tower::ServiceExt;
+    use tempfile::tempdir;
+
+    fn default_args() -> Arc<Args> {
+        Arc::new(Args {
+            target_dir: "/tmp".into(),
+            bind: "127.0.0.1:3000".parse().unwrap(),
+        })
+    }
+
+    #[tokio::test]
+    async fn test_index() {
+        let app = app(default_args());
+
+        let response = app
+            .oneshot(Request::builder().uri("/").body(Body::empty()).unwrap())
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+        assert_eq!(
+            response.headers().get(CONTENT_TYPE).unwrap(),
+            "text/html; charset=utf-8"
+        );
+    }
+
+    #[tokio::test]
+    async fn test_scripts_js() {
+        let app = app(default_args());
+
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .uri("/scripts.js")
+                    .body(Body::empty())
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+        assert_eq!(
+            response.headers().get(CONTENT_TYPE).unwrap(),
+            "application/javascript"
+        );
+    }
+
+    #[tokio::test]
+    async fn test_upload() {
+        let tmp_dir = tempdir().unwrap();
+        let args = Arc::new(Args {
+            target_dir: tmp_dir.path().to_path_buf(),
+            bind: "127.0.0.1:3000".parse().unwrap(),
+        });
+        let app = app(args);
+        let response = app
+            .oneshot(
+                Request::builder()
+                    .method(axum::http::Method::POST)
+                    .uri("/")
+                    .header(CONTENT_TYPE, "multipart/form-data;boundary=95685543938383789682253523760123")
+                    .body(Body::from(
+                            "--95685543938383789682253523760123\r\n\
+                            Content-Disposition: form-data; name=\"file\"; filename=\"test.txt\"\r\n\
+                            Content-Type: text/plain\r\n\
+                            \r\n\
+                            hello\
+                            \r\n\
+                            --95685543938383789682253523760123--"
```

Comment:
During development of this I noticed that our error-handling is quite bare-bones with all these unwraps :wink: 

Can also be reproduced with `httpie/curl`:
```
http --multipart localhost:3001 file='foo'

http: LogLevel.ERROR: ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response')) while doing a POST request to URL: http://localhost:3001/
```